### PR TITLE
Sdk 691/init session forced

### PR DESCRIPTION
**Pull request by bklastaitis-branch**
## Reference
SDK-881 -- Remove branch_force_new_session once it's been used
SDK-888 -- Create an API that disables auto session initialization until some signal is given
SDK-887 -- Create an API that disables auto-session-initialization
SDK-851 -- Make Android plugin type/version api match iOS
SDK-756 -- ServerRequestGetCPID can potentially throw NPE
SDK-691 -- SessionBuilder + Fix initSessionForced 
SDK-672 -- Investigate isActivityCreatedAndLaunched_ 

## Description
Too many ticket, need to read individual Jira tickets for descriptions, if needed I will jump in to explain individual code snippets and which ticket it relates to.

## Testing Instructions
Bunch of integration tests are needed, also possible to build the SDK and try using certain features (like delayed initiualization) on the testbed app. 

## Risk Assessment [`HIGH` || `MEDIUM` || `LOW`]
HIGH

- [ ] I, the PR creator, have tested — integration, unit, or otherwise — this code.

## Reviewer Checklist (To be checked off by the reviewer only)

- [ ] JIRA Ticket is referenced in PR title.
- Correctness & Style
    - [ ] Conforms to [AOSP Style Guides](https://source.android.com/setup/contribute/code-style)
    - [ ] Mission critical pieces are documented in code and out of code as needed.
- [ ] Unit Tests reviewed and test issue sufficiently.
- [ ] Functionality was reviewed in QA independently by another engineer on the team.

### CODE REVIEW

File: Branch-SDK-TestBed/src/main/java/io/branch/branchandroiddemo/MainActivity.java [hunk 1]
```diff
@@ -358,8 +358,8 @@ public void onClick(View view) {
                     return;
                 }
 //                intent.setData(Uri.parse(shortURL));
-                intent.putExtra(Defines.Jsonkey.AndroidPushNotificationKey.getKey(),shortURL);
-                intent.putExtra(Defines.Jsonkey.ForceNewBranchSession.getKey(), true);
+                intent.putExtra(Defines.IntentKeys.BranchURI.getKey(),shortURL);
```
Comment (jdee) [inline Branch-SDK-TestBed/src/main/java/io/branch/branchandroiddemo/MainActivity.java]: Might want a space after the comma, though there wasn't one in the original.

File: Branch-SDK/src/main/java/io/branch/referral/Branch.java [hunk 2]
```diff
@@ -479,6 +473,27 @@ public static void disableDebugMode() {
...
+    /**
+     * Temporarily disables auto session initialization until user initializes themselves.
+     *
+     * Context: Branch expects session initialization to be started in LauncherActivity.onStart(),
+     * if session initialization has not been started/completed by the time ANY Activity resumes,
+     * Branch will auto-initialize. This allows Branch to keep an accurate count of all app sessions,
+     * including instances when app is launched from a recent apps list and the first visible Activity
+     * is not LauncherActivity.
+     *
+     * However, in certain scenarios users may need delay session initialization (e.g. to asynchronously
```
Comment (jdee) [inline Branch-SDK/src/main/java/io/branch/referral/Branch.java]: may need *to* delay session initialization

File: Branch-SDK/src/main/java/io/branch/referral/Branch.java [hunk 3]
```diff
@@ -479,6 +473,27 @@ public static void disableDebugMode() {
...
+     * retrieve some data that needs to be passed to Branch prior to session initialization). In those
+     * cases, use expectDelayedSessionInitialization() to temporarily disable auto self initialization.
+     * Once user initializes the session themselves, the flag will be reset and auto session initialization
```
Comment (jdee) [inline Branch-SDK/src/main/java/io/branch/referral/Branch.java]: Once *the* user initializes

File: Branch-SDK/src/main/java/io/branch/referral/Branch.java [hunk 4]
```diff
@@ -2551,13 +2466,17 @@ private void insertRequestAtFront(ServerRequest req) {
...
+
+    private void initializeSession(final BranchReferralInitListener callback, int delay) {
         if ((prefHelper_.getBranchKey() == null || prefHelper_.getBranchKey().equalsIgnoreCase(PrefHelper.NO_STRING_VALUE))) {
             setInitState(SESSION_STATE.UNINITIALISED);
             //Report Key error on callback
             if (callback != null) {
                 callback.onInitFinished(null, new BranchError("Trouble initializing Branch.", BranchError.ERR_BRANCH_KEY_INVALID));
             }
-            PrefHelper.Debug("Warning: Please enter your branch_key in your project's res/values/strings.xml!");
+            PrefHelper.Debug("Warning: Please enter your branch_key in your project's manifest");
```
Comment (jdee) [inline Branch-SDK/src/main/java/io/branch/referral/Branch.java]: :+1: Noticed this a few times.

File: Branch-SDK/src/main/java/io/branch/referral/ServerRequestGetCPID.java [hunk 5]
```diff
@@ -2,6 +2,8 @@
 
 import android.content.Context;
 
+import androidx.annotation.NonNull;
```
Comment (jdee) [inline Branch-SDK/src/main/java/io/branch/referral/ServerRequestGetCPID.java]: Does this mean this version of the SDK will require androidx? Does that indicate a new major revision?

Comment (bklastaitis-branch) [reply Branch-SDK/src/main/java/io/branch/referral/ServerRequestGetCPID.java]: hmm, androidx was introduced in v4.0.0 though. Has this been causing problems to React Native < 0.60? I though it was working and we already made a release with v4.3.2?

**Pull request merged**